WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasTwoColumnDropDown.lua

```diff
@@ -0,0 +1,210 @@
+        widget:DoLayout()
+      else
+        widget.firstDropdown:SetRelativeWidth(1)
+        widget.secondDropDown.hideMe = true
+        widget:DoLayout()
+        widget:Fire("OnValueChanged", treeValue)
+      end
+    end
+
+    firstDropdown.OnFirstDropdownValueChanged = OnFirstDropdownValueChanged
+
+    local OnSecondDropdownValueChanged = function(self, event, value)
+      local displayName1 = widget.userdata.firstList[widget.firstDropdown:GetValue()]
+      local displayName2 = widget.userdata.secondList[value]
+
+      local treeValue = widget.userdata.tree[displayName1]
+      if not treeValue then
+        return
+      end
+      treeValue = treeValue[displayName2]
+      if not treeValue then
+        return
+      end
+      widget:Fire("OnValueChanged", treeValue)
+    end
+
+    firstDropdown:SetCallback("OnValueChanged", OnFirstDropdownValueChanged)
+    secondDropDown:SetCallback("OnValueChanged", OnSecondDropdownValueChanged)
+  end,
+  ["OnRelease"] = function(self)
+  end,
+  ["SetLabel"] = function(self, ...)
+    self.firstDropdown:SetLabel(...)
+  end,
+  ["SetValue"] = function(self, value)
+    for displayName, treeValue in pairs(self.userdata.tree) do
+      if treeValue == value then
+        self.firstDropdown:SetRelativeWidth(1)
+        self.secondDropDown.hideMe = true
+        self:DoLayout()
+        self.firstDropdown:SetValue(tIndexOf(self.userdata.firstList, displayName))
+        return
+      elseif type(treeValue) == "table" then
+        for displayName2, key in pairs(treeValue) do
+          if (key == value) then
+            self.firstDropdown:SetRelativeWidth(0.5)
+            self.secondDropDown:SetRelativeWidth(0.5)
+            self.secondDropDown.hideMe = false
+            self:DoLayout()
+            local index = tIndexOf(self.userdata.firstList, displayName);
+            self.firstDropdown:SetValue(index)
+            self.firstDropdown:OnFirstDropdownValueChanged("", index)
+            self.secondDropDown:SetValue(tIndexOf(self.userdata.secondList, displayName2))
+            return
+          end
+        end
+      end
+    end
+    self.firstDropdown:SetRelativeWidth(1)
+    self.secondDropDown.hideMe = true
+    self:DoLayout()
+    self.firstDropdown:SetValue(nil)
+  end,
+  ["GetValue"] = function(self)
+    local displayName1 = self.userdata.firstList[self.firstDropdown:GetValue()]
+
+    local treeValue = self.userdata.tree[displayName1]
+    if not treeValue then
+      return nil
+    end
+    if type(treeValue) ~= "table" then
+      return treeValue
+    end
+
+    local displayName2 = self.userdata.secondList[self.secondDropDown:GetValue()]
+    treeValue = treeValue[displayName2]
+    if not treeValue then
+      return nil
+    end
+    return treeValue
+  end,
+  ["SetList"] = function(self, list, order, itemType)
+    local tree = {}
+    for key, displayName in pairs(list) do
+      local pos = string.find(displayName, '/', 1, true)
```

Comment:
Hmm. This is a bit magical to me, and also exposes us to an (admittedly unlikely) issue with translations. If the name of e.g. `Glow` gets translated to contain a slash, then this could break the SetList method at least visually.

Perhaps instead of encoding the structure of a property change in the displayName, we could create a table in each property which contains the value for the first dropdown, and a value for the second (if it exists).

Reply:
What I want to avoid is having the properties be a tree structure, as that makes the whole handling everywhere much more complex. I thik I might be able to cope with displayName be a array though.